storage/tests/StorageManager_unittest.cpp

```diff
@@ -205,46 +205,4 @@ TEST(StorageManagerTest, DifferentNUMANodeBlobTestWithEviction) {
 }
 #endif  // QUICKSTEP_HAVE_LIBNUMA
 
-// Trigger an eviction from the same shard in StorageManager's
```

Comment:
Also, I noticed while refactoring, that when a new block is created (`StorageManager::createBlock`), that this process does not check if we are exceeding our maximum buffer pool size. (previous to this PR, it did not check).

Do we want to change this?

There's a convenient spot in `StorageManager::createBlock` where we have a write lock on the buffer pool:
```c++
  {
    // Lock 'blocks_shared_mutex_' as briefly as possible to insert an entry
    // for the new block, after the block has already been constructed.
    SpinSharedMutexExclusiveLock<false> write_lock(blocks_shared_mutex_);

    // Because block IDs are generated by atomically incrementing block_index_,
    // there should never be collisions.
    DCHECK(blocks_.find(new_block_id) == blocks_.end());
    blocks_[new_block_id] = new_block_handle;

    // We could add this line:
    makeRoomForBlock(0);
    // Which would evict blocks until we go below the system-defined buffer pool size.
```